src/mpihdfio.cxx

```diff
@@ -128,14 +82,16 @@ void MPIDomainDecompositionHDF(Options &opt){
 void MPINumInDomainHDF(Options &opt)
 {
     if (NProcs==1) return;
-    MPIDomainExtentHDF(opt);
-    MPIInitialDomainDecomposition(opt);
-    MPIDomainDecompositionHDF(opt);
+    LOG(debug) << "Determining number of particles in MPI domain";
+    if (opt.cellnodeids.size() == 0) {
+        MPIDomainExtentHDF(opt);
+        MPIInitialDomainDecomposition(opt);
+        MPIDomainDecompositionHDF(opt);
+    }
 
     Int_t i,j,k;
     unsigned long long n,nchunk;
     char buf[2000];
-    MPI_Status status;
```

Comment:
Again, from this line onwards in this function it looks like all changes are unrelated to the mesh/mpi decomposition fixes, is that correct?